src/datamigration/azext_datamigration/manual/custom.py

```diff
@@ -0,0 +1,245 @@
+
+    return defaultOutputPath
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Register Sql Migration Service on IR command Implementation.
+# -----------------------------------------------------------------------------------------------------------------
+def datamigration_register_ir(auth_key,
+                              ir_path=None):
+
+    osPlatform = platform.system()
+
+    if not osPlatform.__contains__('Windows'):
+        raise CLIError("This command cannot be run in non-windows environment. Please run this command in Windows environement")
+
+    if not is_user_admin():
+        raise CLIError("Failed: You do not have Administrator rights to run this command. Please re-run this command as an Administrator!")
+    validate_input(auth_key)
+    if ir_path is not None:
+        if not os.path.exists(ir_path):
+            raise CLIError(f"Invalid gateway path : {ir_path}. Please provide a valid Integration Runtime MSI path")
+        install_gateway(ir_path)
+
+    register_ir(auth_key)
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to check whether the command is run as admin.
+# -----------------------------------------------------------------------------------------------------------------
+def is_user_admin():
+
+    try:
+        isAdmin = os.getuid() == 0
+    except AttributeError:
+        isAdmin = ctypes.windll.shell32.IsUserAnAdmin() != 0
+
+    return isAdmin
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to validate key input.
+# -----------------------------------------------------------------------------------------------------------------
+def validate_input(key):
+    if key == "":
+        raise CLIError("Failed: IR Auth key is empty. Please provide a valid auth key.")
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to check whether SHIR is installed or not.
+# -----------------------------------------------------------------------------------------------------------------
+def check_whether_gateway_installed(name):
+
+    import winreg
+    # Connecting to key in registry
+    accessRegistry = winreg.ConnectRegistry(None, winreg.HKEY_LOCAL_MACHINE)
+
+    # Get the path of Installed softwares
+    accessKey = winreg.OpenKey(accessRegistry, r"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall")
+
+    for i in range(0, winreg.QueryInfoKey(accessKey)[0]):
+
+        installedSoftware = winreg.EnumKey(accessKey, i)
+        installedSoftwareKey = winreg.OpenKey(accessKey, installedSoftware)
+        try:
+            displayName = winreg.QueryValueEx(installedSoftwareKey, r"DisplayName")[0]
+            if name in displayName:
```

Comment:
you are checking for subset of displayName here

Reply:
Yes. because display name can be Microsoft Integration Runtime or Microsoft Integration Runtime Preview.